src/Microsoft.AspNetCore.Mvc.Core/RequireHttpsAttribute.cs

```diff
@@ -76,15 +83,15 @@ protected virtual void HandleNonHttpsRequest(AuthorizationFilterContext filterCo
                     host = new HostString(host.Host);
                 }
 
-                var newUrl = string.Concat(
-                    "https://",
-                    host.ToUriComponent(),
-                    request.PathBase.ToUriComponent(),
-                    request.Path.ToUriComponent(),
-                    request.QueryString.ToUriComponent());
+                var newUrl = UriHelper.Encode(
```

Comment:
I thought it made sense to utilize the `UriHelper` from HttpAbstractions when I was through here anyway :smile: